app/controllers/admin/api_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::ApiController < ApplicationController
+  before_filter :admin_only
+  before_filter :check_for_cancel, only: [:create, :update]
+
+  def index
```

Comment:
OK, so this is not a code error, but my strong suggestion is that you bake in right away a super simple search for the index -- I'm talking just find by name with % variables -- so the admin doesn't have to page through API keys looking for the one they need in the inevitable case soon to come when we have dozens. (I cannot tell you how many times I wish I had done this in the challenge signups sob.) It would only be a few more lines of code and radically simplify key management.